clangd/index/ClangdIndexerImpl.cpp

```diff
@@ -87,7 +89,34 @@ class TranslationUnitToBeIndexedCollector {
       return;
     }
 
+    bool Found = false;
```

Comment:
I think this block of code could be simplified a bit. It also doesn't handle if the excluded path has a trailing / or if it has dots (../../foo).

```
  bool isExcluded(PathRef CurrentPath) {
    return std::any_of(ExclusionList.begin(), ExclusionList.end(), [this, &CurrentPath](const Path& ExlcudedPath) {
      SmallString<10> FullPath(ExlcudedPath);
      if (llvm::sys::path::is_relative(FullPath)) {
        llvm::sys::fs::make_absolute(RootPath, FullPath);
        FullPath = RootPath;
        if (!llvm::sys::path::is_separator(FullPath.back()))
          FullPath += llvm::sys::path::get_separator();
        FullPath += ExlcudedPath;
      }

      llvm::sys::path::remove_dots(FullPath, true);

      return FullPath == CurrentPath;
    });
  }
```